src/ucp/wireup/select.c

```diff
@@ -2142,29 +2141,21 @@ static ucs_status_t ucp_wireup_select_set_locality_flags(
             continue;
         }
 
-        dev_addr = ucs_malloc(wiface->attr.device_addr_len, "ucp_tmp_dev_addr");
-        if (dev_addr == NULL) {
-            ucs_error("failed to allocated device address of size %zu",
-                      wiface->attr.device_addr_len);
-            return UCS_ERR_NO_MEMORY;
-        }
-
-        status = uct_iface_get_device_address(wiface->iface, dev_addr);
-        if (status != UCS_OK) {
-            ucs_free(dev_addr);
-            return status;
-        }
-
         ucp_unpacked_address_for_each(ae, select_params->address) {
-            if ((wiface->attr.device_addr_len == ae->dev_addr_len) &&
-                !memcmp(ae->dev_addr, dev_addr, ae->dev_addr_len)) {
+            if (wiface->attr.device_addr_len != ae->dev_addr_len) {
```

Comment:
maybe better to do this check in `uct_base_iface_is_reachable_v2` itself? Otherwise it may wrongly initialize `is_self` if two addresses have different length, but their content is partially the same

Reply:
Currently `is_reachable_v2` does not have addr_len argument, so we can't perform this comparison there.